### ISSUE TITLE
Support for Door input

### ISSUE THREAD

**Issue by micahdanger** [michaelfranzl/grbl-streamer#4]
I have my GRBL controller (Blackbox from OpenBuilds) wired with an e-stop for the Door input. I get the following error when I trigger that input:

ValueError: could not convert string to float: 'Check Door'

I've added to the parsing of the line from the queue in the _onread method the following:
```
elif "Check Door" in line:
    self.cmode = "Door"
    self._callback("on_stateupdate", self.cmode, self.cmpos, self.cwpos)
    self._callback("on_read", line)
    self._callback("on_alarm", line)
      
elif "Caution: Unlocked" in line:
    self._callback("on_stateupdate", self.cmode, self.cmpos, self.cwpos)
    self._callback("on_read", line)
```
just after:

```
elif line == "ok":
    self._handle_ok()
```

This has alleviated the issue for me, but as I haven't wholly wrapped my head around this code, I'm not sure if that's the right fix. Let me know what you think. I've also have never opened an issue on Github. I hope this is the proper way to do it. Cheers.

**Comment by HendrikRoehm** [comment]
Hey @micahdanger , I had a similar issue when getting:
```
[MSG:'$H'|'$X' to unlock]
```
If your response from the grbl controller is also a msg with some text body, the parsing can be fixed in general by using:
```python
elif re.match("^\[MSG:.*", line):
    self._callback("on_read", line)
```
at the position, you mentioned. Maybe some other callbacks should be called as well, I do not know.

**Comment by michaelfranzl** [comment]
I haven't thoroughly checked, but it seems that this issue is now resolved with #7 . If not, please feel free to re-open.
